### DISCUSSION TITLE
ssh part of git for windows version 2.39 hangs

### DISCUSSION THREAD

**Discussion by user3477071** [git-for-windows/git#4385] [Q&A]
Using this version of git for windows
```
$ git --version --build-options
git version 2.39.2.windows.1
cpu: x86_64
built from commit: a82fa99b36ddfd643e61ed45e52abe314687df67
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon

```

Version of ssh that comes along with this distribution
```
$ ssh -V
OpenSSH_9.1p1, OpenSSL 1.1.1s  1 Nov 2022
```

For some reason the ssh command which comes with git 2.39 hangs in my windows 10 desktop
```
$ ssh -vvv <remote_host>
OpenSSH_9.1p1, OpenSSL 1.1.1s  1 Nov 2022
debug1: Reading configuration data /etc/ssh/ssh_config
<hang>
```
for all remote hosts.

On the same desktop, I have another version of git for windows
```
$ git --version --build-options
git version 2.33.0.windows.2
cpu: x86_64
built from commit: 8735530946cced809cc6cc4c2ca3b078cdb3dfc8
sizeof-long: 4
sizeof-size_t: 8
shell-path: /bin/sh
feature: fsmonitor--daemon
```

whose ssh
```
$ ssh -V
OpenSSH_8.7p1, OpenSSL 1.1.1k  25 Mar 2021 
```
works fine for the same <remote_host>(and all other remote hosts as well) without any hang.

Need assistance on how I can get the hang fixed and use the ssh which comes as part of version 2.39

**Comment by dscho** [comment]
The best idea would be to:

1. verify that the newest Git for Windows [snapshot](https://wingit.blob.core.windows.net/files/index.html) is also broken (if not, just use it).
2. if it is broken, bisect through the snapshots to find out which one is the first one that is broken (you can use the PortableGit version, that way you can keep your current installed Git for Windows version that works for you).
3. once the first broken version is identified, compare the `/etc/package-versions.txt` files from this version relative to the preceding one.
4. replace files corresponding to the differences (e.g. `usr\bin\msys-2.0.dll` if the `msys2-runtime` package was changed, or `usr\bin\ssh.exe` if the `openssh` package was changed) in the broken version with the files from the non-broken version and see whether this fixes the regression.
